File: src/master/syscall/types/bot.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

/// A shard connection (for bot statistics)
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ShardConn {
    /// The status of the shard connection
    pub status: String,
    /// The real latency of the shard connection in milliseconds
    pub latency: f64,
}

/// Status of all shards and other info like total guilds/users
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BotStatus {
    /// A map of shard group ID to shard connection information
    pub shard_conns: HashMap<u32, ShardConn>,
    /// The total number of guilds the bot is connected to
    pub total_guilds: u64,
    /// The total number of users
    pub total_users: u64,
}

File: devutils/willow/src/lib/msyscall/types/bot.ts
export interface ShardConn {
  /** The status of the shard connection */
  status: string;
  /** The real latency of the shard connection in milliseconds */
  latency: number;
}

export interface BotStatus {
  /** A map of shard group ID to shard connection information */
  shard_conns: Record<number, ShardConn>;
  /** The total number of guilds the bot is connected to */
  total_guilds: number;
  /** The total number of users */
  total_users: number;
  /** The current uptime of the bot process in seconds */
  uptime: number;
}
